package/kedro_viz/api/graphql.py

```diff
@@ -42,111 +42,97 @@
-        gitSha="892372937",
+    git_data = run_blob.get("git")
+    run = Run(
+        id=ID(run_id),
+        author="",
+        gitBranch="",
+        gitSha=git_data.get("commit_sha") if git_data else None,
+        bookmark=False,
+        title=run_blob["session_id"],
         notes="",
-        runCommand="kedro run",
-    )
-    details = RunDetails(id=ID("123"), name="name", details="{json:details}")
-
-    return Run(
-        id=ID("123"),
-        bookmark=True,
-        timestamp="2021-09-08T10:55:36.810Z",
-        title="Sprint 5",
-        metadata=metadata,
-        details=details,
+        timestamp=run_blob["session_id"],
+        runCommand=run_blob["cli"]["command_path"],
     )
+    return run
 
 
-def get_runs() -> List[Run]:
-    """Placeholder for the proper method.
-    Get all runs from the session store.
+def get_run(run_id: ID) -> Run:
+    """Get a run by id from the session store.
+
+    Args:
+        run_id: ID of the run to fetch
 
     Returns:
-        list of Run objects
+        Run object
     """
-    return [get_run(ID("123"))]
-
-
-@strawberry.type
-class RunModelGraphQLType:
-    """RunModel format to return to the frontend"""
-
-    id: str
-    blob: str
+    session = data_access_manager.db_session
+    run_data = session.query(RunModel).filter(RunModel.id == run_id).first()
+    return format_run(run_data.id, json.loads(run_data.blob))
 
 
-def get_all_runs() -> typing.List[RunModelGraphQLType]:
-    """Gets all runs from the session store
+def get_runs() -> List[Run]:
+    """Get all runs from the session store.
 
     Returns:
         list of Run objects
-
     """
-    return [
-        RunModelGraphQLType(id=kedro_session.id, blob=kedro_session.blob)
-        for kedro_session in data_access_manager.db_session.query(RunModel).all()
-    ]
+    runs = []
+    session = data_access_manager.db_session
+    for run_data in session.query(RunModel).all():
+        run = format_run(run_data.id, json.loads(run_data.blob))
+        runs.append(run)
+    return runs
 
 
 @strawberry.type
 class Run:
-    """Run object format to return to the frontend"""
+    """Run object format"""
 
     id: ID
-    bookmark: bool
-    timestamp: str
     title: str
-    metadata: RunMetadata
-    details: RunDetails
-
-
-@strawberry.type
-class RunMetadata:
-    """RunMetadata object format"""
-
-    id: ID
-    author: str
-    gitBranch: str
-    gitSha: str
-    notes: str
-    runCommand: str
+    timestamp: str
+    author: Optional[str]
+    gitBranch: Optional[str]
+    gitSha: Optional[str]
+    bookmark: Optional[bool]
+    notes: Optional[str]
+    runCommand: Optional[str]
 
 
 @strawberry.type
-class RunDetails:
-    """RunDetails object format"""
+class TrackingDataSet:
+    """TrackingDataSet object to structure tracking data for a Run."""
 
-    id: ID
-    name: str
-    details: str
+    datasetName: Optional[str]
+    datasetType: Optional[str]
+    data: Optional[str]
```

Comment:
Why are all these optional?

Reply:
This was still my doing, but matches the schema on the frontend:
```
type TrackingDataset {
    datasetName: String
    datasetType: String
    data: JSONObject
  }
```